Stencil Params: Add Stencil Compare and Write Mask Material Parameters

gkjohnson opened the pull request:
Fix #17419

Changes `stencilMask` to `stencilFuncMask` and adds `stencilWriteMask` (corresponding to the [stencilMask](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/stencilMask) function).

For the moment the setter and getter for `stencilMask` will log a warning they are used.

gkjohnson committed:
Add stencil write mask field

gkjohnson committed:
Add removal warning

gkjohnson committed:
Update docs

gkjohnson committed:
Update zh docs

gkjohnson committed:
Update copy function

gkjohnson committed:
linting fix

Mugen87 commented on src/materials/Material.js:
```diff
@@ -41,14 +41,33 @@ function Material() {
+		}
+
+	} );
+
```
Stuff like that normally goes to `three.legacy.js`.

Mugen87 commented on src/materials/Material.js:
```diff
@@ -362,9 +381,10 @@ Material.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 		this.stencilFunc = source.stencilFunc;
 		this.stencilRef = source.stencilRef;
-		this.stencilMask = source.stencilMask;
+		this.stencilFuncMask = source.stencilFuncMask;
```
It might also be a good idea to enhance `MaterialLoader` so these properties can be actually deserialized.

Mugen87 commented:
It seems you also have to change `Material.toJSON()`, see

https://github.com/mrdoob/three.js/pull/17449/files#diff-f5eb8fcc970231604c15522381d9f59eR282

gkjohnson committed:
update Material.json and add `stencilMask` to THREE.legacy

gkjohnson committed:
Update MaterialLoader

gkjohnson commented:
@Mugen87 Thanks I've updated `toJSON`, moved the property warning to `Three.Legacy.js` and updated `MaterialLoader` to deserialize stencil properties!

mrdoob commented:
Thanks!
